datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -121,5 +183,49 @@ select unnest(array_remove(column1, 12)) from unnest_table;
 5
 6
 
+
+## Unnest in from clause with alias
+query I
+select * from unnest([1,2]) as t;
+----
+1
+2
+
+query I
+select a from unnest([1,2]) as t(a);
+----
+1
+2
+
+
+## Unnest in from clause with offset is not supported
+query error DataFusion error: This feature is not implemented: UNNEST table factor with offset is not supported yet
+select * from unnest([1,2]) with offset;
+
+query error DataFusion error: This feature is not implemented: UNNEST table factor with offset is not supported yet
+select * from unnest([1,2]) with offset offset_alias;
+
+
+## More complex cases
+query I
+select * from unnest([1,2,(select sum(column3) from unnest_table)]);
+----
+1
+2
+6
+
+query II??I rowsort
+select * from unnest([1]), unnest([2, 3]), unnest_table;
```

Comment:
This one is incorrect.
If a is [1,2,3] and [4,5] (2 rows)
```
postgres=# select * from unnest(ARRAY[1,2,3]), unnest(ARRAY[4,5]) as b2, t1;
 unnest | b2 |    a    
--------+----+---------
      1 |  4 | {1,2,3}
      2 |  4 | {1,2,3}
      3 |  4 | {1,2,3}
      1 |  5 | {1,2,3}
      2 |  5 | {1,2,3}
      3 |  5 | {1,2,3}
      1 |  4 | {4,5}
      2 |  4 | {4,5}
      3 |  4 | {4,5}
      1 |  5 | {4,5}
      2 |  5 | {4,5}
      3 |  5 | {4,5}
(12 rows)
```

Similar for Duckdb, a is [1,2,3] and [10, 20]
```
D select * from unnest([1,2,3]), unnest([4,5]) as b2, t1;
┌──────────────────────────┬───────────────────────┬───────────┐
│ main.list_value(1, 2, 3) │ main.list_value(4, 5) │     a     │
│          int32           │         int32         │  int32[]  │
├──────────────────────────┼───────────────────────┼───────────┤
│                        1 │                     4 │ [1, 2, 3] │
│                        2 │                     4 │ [1, 2, 3] │
│                        3 │                     4 │ [1, 2, 3] │
│                        1 │                     5 │ [1, 2, 3] │
│                        2 │                     5 │ [1, 2, 3] │
│                        3 │                     5 │ [1, 2, 3] │
│                        1 │                     4 │ [10, 20]  │
│                        2 │                     4 │ [10, 20]  │
│                        3 │                     4 │ [10, 20]  │
│                        1 │                     5 │ [10, 20]  │
│                        2 │                     5 │ [10, 20]  │
│                        3 │                     5 │ [10, 20]  │
├──────────────────────────┴───────────────────────┴───────────┤
│ 12 rows                                            3 columns │
└──────────────────────────────────────────────────────────────┘
```

Reply:
It doesn't seem like a easy fix, you can remove the test or uncomment it.

Reply:
Let me remove it first, thank you very much for pointing it out.

Reply:
I found that the result of this query is the same as with DuckDB, except for the order of the rows. Did I overlook something else?

In DataFusion:
```sh
DataFusion CLI v36.0.0
❯ create table t1 as values([1,2,3]), ([10, 20]);
0 rows in set. Query took 0.010 seconds.

❯ select * from unnest([1,2,3]), unnest([4,5]) as b2, t1;
+----------------------------------------+-------------------------------+-----------+
| make_array(Int64(1),Int64(2),Int64(3)) | make_array(Int64(4),Int64(5)) | column1   |
+----------------------------------------+-------------------------------+-----------+
| 1                                      | 4                             | [1, 2, 3] |
| 1                                      | 4                             | [10, 20]  |
| 1                                      | 5                             | [1, 2, 3] |
| 1                                      | 5                             | [10, 20]  |
| 2                                      | 4                             | [1, 2, 3] |
| 2                                      | 4                             | [10, 20]  |
| 2                                      | 5                             | [1, 2, 3] |
| 2                                      | 5                             | [10, 20]  |
| 3                                      | 4                             | [1, 2, 3] |
| 3                                      | 4                             | [10, 20]  |
| 3                                      | 5                             | [1, 2, 3] |
| 3                                      | 5                             | [10, 20]  |
+----------------------------------------+-------------------------------+-----------+
12 rows in set. Query took 0.014 seconds.
```
And in DuckDB:
```sh
v0.9.3-dev3269 8de2d291d8

D create table t1 as values([1,2,3]), ([10, 20]);
D select * from unnest([1,2,3]), unnest([4,5]) as b2, t1;
┌──────────────────────────┬───────────────────────┬───────────┐
│ main.list_value(1, 2, 3) │ main.list_value(4, 5) │   col0    │
│          int32           │         int32         │  int32[]  │
├──────────────────────────┼───────────────────────┼───────────┤
│                        1 │                     4 │ [1, 2, 3] │
│                        2 │                     4 │ [1, 2, 3] │
│                        3 │                     4 │ [1, 2, 3] │
│                        1 │                     5 │ [1, 2, 3] │
│                        2 │                     5 │ [1, 2, 3] │
│                        3 │                     5 │ [1, 2, 3] │
│                        1 │                     4 │ [10, 20]  │
│                        2 │                     4 │ [10, 20]  │
│                        3 │                     4 │ [10, 20]  │
│                        1 │                     5 │ [10, 20]  │
│                        2 │                     5 │ [10, 20]  │
│                        3 │                     5 │ [10, 20]  │
├──────────────────────────┴───────────────────────┴───────────┤
│ 12 rows                                            3 columns │
└──────────────────────────────────────────────────────────────┘
```

Reply:
You are right, only order is different.

Reply:
In this case, postgres and duckdb has the same ordering, I recommend we should have the same ordering too

Reply:
I think the order will not affect correctness if it isn't explicitly specified, due to the parallel execution across multiple partitions, various execution strategies, and so on. What do you think?

Reply:
Actually, the execution results of many queries in DatauFusion are non-deterministic. For example #9368

Reply:
I know some queries are non-deterministic and the result may be different per run. The unnest example here is quite different than the previous one that we get the same result every time. I think the ordering is well-defined here (but I did not search any document about how postgres and duckdb work), why do you think the ordering is due to the multiple partitions

Reply:
> but I did not search any document about how postgres and duckdb work

If the ordering is not specified anywhere, then it is fine.

Reply:
It might due to something else like join order. I think there might be performance issues, but the correctness should be fine in my opinion.

Reply:
Maybe file an issue about it to see whether it is issue in cross join or not

Reply:
> Maybe file an issue about this to track whether it is an issue in cross-join or not

That's a good idea, let me do some further investigation to make this question clearer.

Reply:
example here, not sure what happen in physical plan
```
statement ok
create table t1 as values
  ([1,2,3], 1),
  ([4,5,6], 2)
;

statement ok
create table t2 as values
  (10),
  (20)
;

query ?II
select * from t1 cross join t2;
----
[1, 2, 3] 1 10
[4, 5, 6] 2 10
[1, 2, 3] 1 20
[4, 5, 6] 2 20

query TT
explain select * from t1 cross join t2;
----
logical_plan
CrossJoin:
--TableScan: t1 projection=[column1, column2]
--TableScan: t2 projection=[column1]
physical_plan
ProjectionExec: expr=[column1@1 as column1, column2@2 as column2, column1@0 as column1]
--CrossJoinExec
----MemoryExec: partitions=1, partition_sizes=[1]
----MemoryExec: partitions=1, partition_sizes=[1]

query II
select * from unnest([1,2,3]) as b2, t2;
----
1 10
1 20
2 10
2 20
3 10
3 20

query TT
explain select * from unnest([1,2,3]) as b2, t2;
----
logical_plan
CrossJoin:
--SubqueryAlias: b2
----Unnest: make_array(Int64(1),Int64(2),Int64(3))
------Projection: List([1, 2, 3]) AS make_array(Int64(1),Int64(2),Int64(3))
--------EmptyRelation
--TableScan: t2 projection=[column1]
physical_plan
CrossJoinExec
--UnnestExec
----ProjectionExec: expr=[[1, 2, 3] as make_array(Int64(1),Int64(2),Int64(3))]
------PlaceholderRowExec
--MemoryExec: partitions=1, partition_sizes=[1]
```

Reply:
The result is different from DuckDB's; perhaps there is a need to further investigate the implementation of cross join for both.